### DISCUSSION TITLE
docker-gentoo-distcc cross compiling?

### DISCUSSION THREAD

**Discussion by elsni** [KSmanis/docker-gentoo-distcc#37] [General]
Does the image support cross compiling for Gentoo?
Lets say my docker-host is amd64 and the gentoo client (which is painfully slow in compiling) is i686, is there a chance that it works?
Can also an arm64 machine (Mac with M1) work as another distcc node for i686?

**Comment by KSmanis** [comment]
No, the image does not support cross-compiling; only distcc is bundled, not crossdev. Nevertheless, you don't need a cross-compiler for an amd64 host to compile x86 binaries. You do need a cross-compiler for an arm64 machine though. Alternatively, you could probably run the 386 image on the arm64 machine using QEMU emulation (`docker run --platform linux/386 ... ksmanis/gentoo-distcc`), but I guess this would be much slower than setting up a proper cross-compiler.

**Comment by elsni** [comment]
Thanks for the quick answer. I'll give the image a try on the amd64 host
